test/start-kubernetes.sh

```diff
@@ -0,0 +1,301 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+TEST_CONFIG=${TEST_CONFIG:-${TEST_DIRECTORY}/test-config.sh}
+DEPLOYMENT_SUFFIX=${DEPLOYMENT_SUFFIX:-govm}
+CLOUD_USER=${CLOUD_USER:-clear}
+CLUSTER=${CLUSTER:-${CLOUD_USER}-${DEPLOYMENT_SUFFIX}}
+DEPLOYMENT_ID=${DEPLOYMENT_ID:-pmem-csi-${CLUSTER}}
+GOVM_YAML=${GOVM_YAML:-$(mktemp --suffix $DEPLOYMENT_ID.yml)}
+REPO_DIRECTORY=${REPO_DIRECTORY:-$(dirname $TEST_DIRECTORY)}
+TEST_DIRECTORY=${TEST_DIRECTORY:-$(dirname $(readlink -f $0))}
+RESOURCES_DIRECTORY=${RESOURCES_DIRECTORY:-${REPO_DIRECTORY}/_work/resources}
+WORKING_DIRECTORY="${WORKING_DIRECTORY:-${REPO_DIRECTORY}/_work/${CLUSTER}}"
+source ${TEST_CONFIG}
+NODES=( $DEPLOYMENT_ID-master
+	$DEPLOYMENT_ID-worker1
+	$DEPLOYMENT_ID-worker2
+	$DEPLOYMENT_ID-worker3)
+CLOUD="${CLOUD:-true}"
+FLAVOR="${FLAVOR:-medium}"
+SSH_KEY="${SSH_KEY:-${RESOURCES_DIRECTORY}/id_rsa}"
+SSH_PUBLIC_KEY="${SSH_KEY}.pub"
+IMAGE_TAG="${IMAGE_TAG:-canary}"
+EFI="${EFI:-true}"
+KVM_CPU_OPTS="${KVM_CPU_OPTS:-\
+		-m 2G,slots=${TEST_MEM_SLOTS:-2},maxmem=34G -smp 4 \
+		-machine pc,accel=kvm,nvdimm=on}"
+EXTRA_QEMU_OPTS="${EXTRA_QWEMU_OPTS:-\
+		-object memory-backend-file,id=mem1,share=${TEST_PMEM_SHARE:-on},\
+		mem-path=/data/nvdimm0,size=${TEST_PMEM_MEM_SIZE:-32768}M \
+		-device nvdimm,id=nvdimm1,memdev=mem1,label-size=${TEST_PMEM_LABEL_SIZE:-2097152} \
+		-machine pc,nvdimm}"
+CLOUD_USER=${CLOUD_USER:-clear}
+CLOUD_IMAGE="${CLOUD_IMAGE:-$(\
+		curl -s https://download.clearlinux.org/image/latest-images |\
+		awk '/cloud.img/ {gsub(".xz",""); print $0}')}"
+IMAGE_URL=${IMAGE_URL:-http://download.clearlinux.org/image/${CLOUD_IMAGE}.xz}
+SSH_TIMEOUT=60
+SSH_ARGS="-oIdentitiesOnly=yes -oStrictHostKeyChecking=no \
+	-oUserKnownHostsFile=/dev/null -oLogLevel=error \
+	-i ${SSH_KEY}"
+CREATE_REGISTRY=${CREATE_REGISTRY:-false}
+CHECK_SIGNED_FILES=${CHECK_SIGNED_FILES:-true}
+
+function error_handler(){
+	local line="${1}"
+	echo  "Error unning the ${BASH_COMMAND} on function ${FUNCNAME[1]} at line ${line}"
+	delete_vms
+}
+
+function download_image(){
+    pushd $RESOURCES_DIRECTORY &>/dev/null
+	if [ -e "$CLOUD_IMAGE" ]; then
+		echo "$CLOUD_IMAGE found, skipping download"
+	else
+		case $CLOUD_USER in
+			clear)
+				ca_url="https://cdn.download.clearlinux.org/"
+				ca_url+="releases/${IMAGE_URL//[^0-9]}/"
+				ca_url+="clear/ClearLinuxRoot.pem"
+				echo "Downloading ${CLOUD_IMAGE}.xz image"
+				curl -O ${IMAGE_URL}
+				curl -s -O ${IMAGE_URL}-SHA512SUMS
```

Comment:
downloaded SHA512 and ca-related files are only needed inside next 'if', so download cmds can be moved there?

Reply:
Thanks, I moved the SHA512  and ca-related files there